RoslynSecurityGuard.Test/Tests/OutputCacheAnnotationAnalyzerTest.cs

```diff
@@ -93,20 +92,376 @@ public ActionResult Index()
+        }
+
+        [TestMethod]
+        public void DetectAnnotation5()
+        {
+            var test = @"
+using System;
+using System.Web.Mvc;
+
+public class MyController : Controller
+{
+    [Authorize]
+    public virtual ActionResult Index()
+    {
+        return null;
+    }
+}
+
+[OutputCache]
+public class HomeController : MyController
+{
+    public override ActionResult Index()
+    {
+        return View();
+    }
+}
+";
+            var expected = new DiagnosticResult
+            {
+                Id = OutputCacheAnnotationAnalyzer.DiagnosticId,
+                Severity = DiagnosticSeverity.Warning
+            };
+
+            VerifyCSharpDiagnostic(test, expected);
+        }
+
+        [TestMethod]
+        public void DetectAnnotation6()
+        {
+            var test = @"
+using System;
+using System.Web.Mvc;
+
+public abstract class MyController : Controller
+{
+    [Authorize]
+    public abstract ActionResult Index();
+}
+
+[OutputCache]
+public class HomeController : MyController
+{
+    public override ActionResult Index()
+    {
+        return View();
+    }
+}
+";
+            var expected = new DiagnosticResult
+            {
+                Id = OutputCacheAnnotationAnalyzer.DiagnosticId,
+                Severity = DiagnosticSeverity.Warning
+            };
+
+            VerifyCSharpDiagnostic(test, expected);
+        }
+
+        [TestMethod]
+        public void DetectAnnotation7()
+        {
+            var test = @"
+using System;
+using System.Web.Mvc;
+
+public class MyController : Controller
+{
+    [Authorize]
+    public virtual ActionResult Index()
+    {
+        return null;
+    }
+}
+
+public class HomeController : MyController
+{
+    [OutputCache]
+    public override ActionResult Index()
+    {
+        return View();
+    }
+}
+";
+            var expected = new DiagnosticResult
+            {
+                Id = OutputCacheAnnotationAnalyzer.DiagnosticId,
+                Severity = DiagnosticSeverity.Warning
+            };
+
+            VerifyCSharpDiagnostic(test, expected);
+        }
+
+        [TestMethod]
+        public void DetectAnnotation8()
+        {
+            var test = @"
+using System;
+using System.Web.Mvc;
+
+[OutputCache(NoStore = true, Duration = 0, VaryByParam = ""*"")]
+public class MyController : Controller
+{
+}
+
+[OutputCache(NoStore = true, Duration = int.MaxValue, VaryByParam = ""*"")]
+public class HomeController : MyController
+{
+    [Authorize]
+    public ActionResult Index()
+    {
+        return View();
+    }
+}
+";
+            var expected = new DiagnosticResult
+            {
+                Id = OutputCacheAnnotationAnalyzer.DiagnosticId,
+                Severity = DiagnosticSeverity.Warning
+            };
+
+            VerifyCSharpDiagnostic(test, expected);
+        }
+
+// The question is if we want to go so far and detect a derived attribute since the caching logic can be altered
```

Comment:
@h3xstream, I haven't seen it yet. But wanted to keep the commented out tests for the reference. Since it is now in a changeset history do you want me to delete it?